packages/blade/src/components/ActionList/ActionListFooter.tsx

```diff
@@ -0,0 +1,106 @@
+/* eslint-disable @typescript-eslint/prefer-ts-expect-error */
+/* eslint-disable @typescript-eslint/ban-ts-comment */
+
+import React from 'react';
+import styled from 'styled-components';
+import { componentIds } from './componentIds';
+import Box from '~components/Box';
+import type { IconComponent } from '~components/Icons';
+import { useDropdown } from '~components/Dropdown/useDropdown';
+import { isReactNative, makeSize } from '~utils';
+import { Text } from '~components/Typography';
+
+type ActionListFooterProps = {
+  title?: string;
+  leading?: React.ReactNode;
+  trailing?: React.ReactNode;
+};
+
+const StyledActionListFooter = styled(Box)((props) => {
```

Comment:
So I am using styled-components in parts where there are too many event handlers or other props in the component. Also the padding and margin shorthands are not supported on Box right now so it will increase the props.

So with this the 'styling' is separate from the 'logic'. Don't have strong opinion so either will work.